cmd/gitbase/command/server.go

```diff
@@ -146,23 +146,19 @@ func (c *Server) Execute(args []string) error {
 				Fatal("unable to read jaeger environment")
 			return err
 		}
-
 		if cfg.ServiceName == "" {
 			cfg.ServiceName = TracerServiceName
 		}
 
 		logger := &jaegerLogrus{logrus.WithField("subsystem", "jaeger")}
 
-		t, closer, err := cfg.NewTracer(
-			config.Logger(logger),
-		)
-
+		closer, err := cfg.InitGlobalTracer(cfg.ServiceName, config.Logger(logger))
 		if err != nil {
-			logrus.WithField("error", err).Fatal("unable to initialize tracer")
+			logrus.WithField("error", err).Fatal("unable to initialize global tracer")
 			return err
 		}
 
-		tracer = t
+		tracer = opentracing.GlobalTracer()
```

Comment:
Do you need the tracer in this case? Any library can call `opentracing.GlobalTracer()` to get it. I think it's probably a single package that generates a root context for requests.

Reply:
yes, but we pass it to the go-mysql-server, so thanks to this we can use global (or any other tracer) in gitbase without changing anything in mysql package.